components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/spark/algorithms/UnsupervisedSparkModelBuilder.java

```diff
@@ -90,6 +96,9 @@ public MLModel build() throws MLModelBuilderException {
             Workflow workflow = context.getFacts();
             long modelId = context.getModelId();
             ModelSummary summaryModel = null;
+            SortedMap<Integer, String> includedFeatures = MLUtils
```

Comment:
Code to interfaces, not concrete types, to be able to improve maintainability of the code.